lib/modules/platform/gitlab/index.ts

```diff
@@ -697,7 +697,9 @@ export function massageMarkdown(input: string): string {
   let desc = input
     .replace(regEx(/Pull Request/g), 'Merge Request')
     .replace(regEx(/PR/g), 'MR')
-    .replace(regEx(/\]\(\.\.\/pull\//g), '](!');
+    .replace(regEx(/\]\(\.\.\/pull\//g), '](!')
+    // Strip unicode null characters as GitLab markdown does not permit them
+    .replace(regEx(/\u0000/g), "");
```

Comment:
still doesn't work, maybe do a simple string replace instead?

Reply:
Urgh sorry @viceice, something silly going on here :( The test passes if I run it locally in focus mode (changing `it` to `fit` for my newly added test), but fails when run with other tests in the file.

I did a bit of debugging and the call to `desc = smartTruncate(desc, 25000);` is what's setting `desc` to `undefined` in my test case. If I add a `console.log` to the function body for `smartTruncate`, it's not being reached in my test case.

I think the `jest.mock('../utils/pr-body'); const { smartTruncate } = require('../utils/pr-body');` mocks are not isolated properly between the tests. If I comment out the three tests that do that and run the whole file, my change and newly added test pass.

There is an existing `beforeEach` block that's trying to do something to reset the mocks, but I'm not super familiar with jest, and it doesn't seem to be working in this context.

Do you have any pointers?

Reply:
move your test above the all other massageMarkdown tests for now

Reply:
That appears to have worked 👍 Pushed a commit reordering the tests.